views.py

```diff
@@ -30,7 +30,8 @@ def search(request):
         q, suffix = split_query(q)
 
         for lang in all_langs:
-            if lang['two'].startswith(q) or lang['three'].startswith(q):
+            temp_lang = lang['names'] + [lang['two'], lang['three']]
+            if [n for n in temp_lang if n.lower().startswith(q)]:
```

Comment:
i suspect it wont be very optimised for a very very huge list of languages. But for the set of languages we have it is pretty quick.

Reply:
Nit: Explicit (even if longer) variable names are more readable than one character variables. Also using `any` would avoid the training `if` and add readability.
```
if any([code_or_name.lower().startswith(q) for code_or_name in code_or_names])
```